### PR TITLE
Danielle/patch

### PR DESCRIPTION
**Pull request by bardielle**

### CODE REVIEW

**Comment by bardielle** [comment]
@nirarg 
Also We talked about:
1- Remove machine functions from utils.go file
2- The name stub_test is a little bit confusing

File: pkg/managers/vm/machine_scope.go [hunk 1]
```diff
@@ -41,62 +54,69 @@ func newMachineScope(machine *machinev1.Machine, kubernetesClient kubernetesclie
...
+	}
+	userDataByte, ok := userDataSecret.Data[userDataKey]
+	userData := string(userDataByte)
 	virtualMachine := kubevirtapiv1.VirtualMachine{
 		Spec: kubevirtapiv1.VirtualMachineSpec{
+			//Running: &running,
 			RunStrategy: &runAlways,
 			DataVolumeTemplates: []cdiv1.DataVolume{
-				*buildBootVolumeDataVolumeTemplate(machine.GetName(), providerSpec.SourcePvcName, namespace, providerSpec.SourcePvcNamespace),
+				*buildBootVolumeDataVolumeTemplate(s.machine.GetName(), s.machineProviderSpec.SourcePvcName, namespace, s.machineProviderSpec.SourcePvcNamespace),
```
[the remark this answers is not in the document]
Comment (bardielle) [reply pkg/managers/vm/machine_scope.go:104]: What do you mean a valid name?

File: pkg/managers/vm/vm.go [hunk 2]
```diff
@@ -198,9 +213,14 @@ func (m *manager) Exists(machine *machinev1.Machine) (bool, error) {
 		return false, err
 	}
 
+	virtualMachineFromMachine, err := machineScope.machineToVirtualMachine()
```
[the remark this answers is not in the document]
Comment (bardielle) [reply pkg/managers/vm/vm.go:216]: But the namespace is not the machine.namespace....

File: pkg/managers/vm/machine_scope.go [hunk 3]
```diff
@@ -242,7 +268,7 @@ func buildVMITemplate(virtualMachineName string, providerSpec *kubevirtproviderv
 }
 
 func buildBootVolumeName(virtualMachineName string) string {
-	return virtualMachineName + "-bootvolume"
+	return virtualMachineName + "-" + defaultBootVolumeDiskName
```
[the remark this answers is not in the document]
Comment (bardielle) [reply pkg/managers/vm/machine_scope.go:271]: There are multiple places that use the function buildDataVolumeDiskName in order to map the disk to the same volume
Im afraid that if I will change it and the code will determine the suffix someone can change one place and not in the other....

File: pkg/managers/vm/machine_scope.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (nirarg) [reply pkg/managers/vm/machine_scope.go:104]: Not empty, for example

File: pkg/managers/vm/machine_scope.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (nirarg) [reply pkg/managers/vm/machine_scope.go:271]: What I said is to keep those function, but all three functions has to use the new suggested function as common logic

File: pkg/managers/vm/machine_scope.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (bardielle) [reply pkg/managers/vm/machine_scope.go:104]: machine must has a name field - if not the machine-api-operator would be failed

File: pkg/managers/vm/machine_scope.go [hunk 4]
```diff
@@ -151,47 +171,34 @@ func (s *machineScope) setMachineCloudProviderSpecifics(vm *kubevirtapiv1.Virtua
...
 func buildDataVolumeDiskName(virtualMachineName string) string {
-	return virtualMachineName + defaultDataVolumeDiskName
+	return virtualMachineName + "-" + defaultDataVolumeDiskName
+}
+func buildCloudInitVolumeDiskName(virtualMachineName string) string {
+	return virtualMachineName + "-" + defaultCloudInitVolumeDiskName
 }
 
-func buildVMITemplate(virtualMachineName string, providerSpec *kubevirtproviderv1.KubevirtMachineProviderSpec) *kubevirtapiv1.VirtualMachineInstanceTemplateSpec {
+func buildVMITemplate(virtualMachineName string, providerSpec *kubevirtproviderv1.KubevirtMachineProviderSpec, userData string) *kubevirtapiv1.VirtualMachineInstanceTemplateSpec {
```
[the remark this answers is not in the document]
Comment (bardielle) [reply pkg/managers/vm/machine_scope.go:201]: I agree with you about the struct function but getuserData might return an error - so I will not move it inside that function. What do you think?

File: pkg/managers/vm/machine_scope.go [hunk 5]
```diff
@@ -276,27 +302,32 @@ func buildBootVolumeDataVolumeTemplate(virtualMachineName, pvcName, dvNamespace,
...
-	// TODO: NodeInternalIP of VM
+func (s *machineScope) SyncMachineFromVm(vm *kubevirtapiv1.VirtualMachine, vmi *kubevirtapiv1.VirtualMachineInstance) error {
+	// TODO this function need to be removed
+	s.setProviderID(vm)
 
-	s.setProviderID(s.virtualMachine)
-
-	if err := s.setMachineCloudProviderSpecifics(s.virtualMachine); err != nil {
+	// TODO  this function need to be removed
+	if err := s.setMachineCloudProviderSpecifics(vm); err != nil {
```
[the remark this answers is not in the document]
Comment (bardielle) [reply pkg/managers/vm/machine_scope.go:310]: 1 - I moved the comment
2- Changed it to 'setMachineAnnotationsAndLabels'

File: pkg/managers/vm/vm.go [hunk 6]
```diff
@@ -144,51 +156,54 @@ func (m *manager) Update(machine *machinev1.Machine) (resultErr error) {
...
-		return fmt.Errorf("failed to set machine cloud provider specifics: %w", err)
+		return wasUpdated, fmt.Errorf("failed to set machine cloud provider specifics: %w", err)
 	}
 
 	klog.Infof("Updated machine %s", machineScope.getMachineName())
 
-	machineScope.setProviderStatus(updatedVM, conditionSuccess())
-
-	getUpdatedVM, err := m.getVM(machineScope.virtualMachine.GetName(), machineScope.virtualMachine.GetNamespace(), machineScope)
+	getUpdatedVM, err := m.getVM(updatedVM.GetName(), updatedVM.GetNamespace(), machineScope)
```
[the remark this answers is not in the document]
Comment (bardielle) [reply pkg/managers/vm/vm.go:194]: I will write it as a comment - the updatedVM doesn't include things just after updating

**Pull request merged**